modules/openthread/platform/border_agent.c

```diff
@@ -0,0 +1,235 @@
+					.mValueLength = strlen(otbr_vendor_name)},
+				       {.mKey = "mn", .mValue = (uint8_t *)otbr_model_name,
+					.mValueLength = strlen(otbr_model_name)}};
+
+	for (i = 0; i < ARRAY_SIZE(txt_entries); ++i) {
+		const otDnsTxtEntry *entry    = &txt_entries[i];
+		uint8_t key_len   = (uint8_t)strlen(entry->mKey);
+		uint8_t total_len = key_len + 1 + entry->mValueLength; /* +1 for '=' */
+
+		txt_data[(*txt_data_len)++] = total_len;
+		memcpy(txt_data + *txt_data_len, entry->mKey, key_len);
+		*txt_data_len += key_len;
+
+		txt_data[(*txt_data_len)++] = '=';
+		memcpy(txt_data + *txt_data_len, entry->mValue, entry->mValueLength);
+		*txt_data_len += entry->mValueLength;
+	}
+
+}
+
+#if defined(CONFIG_OPENTHREAD_BORDER_AGENT_EPHEMERAL_KEY_ENABLE)
+
+/** Defining functions below as weak allows applications to implement their specific behaviour,
+ *  i.e., custom messages/used print function.
+ */
+void __attribute__((weak)) print_ephemeral_key(const char *ephemeral_key, uint32_t timeout)
+{
+	otCliOutputFormat("\r\n Use this passcode to enable an additional device to administer "
+			  "and manage your Thread network, including adding new devices to it. "
+			  "\r\nThis passcode is not required for an app to communicate with "
+			  "existing devices on your Thread network.");
+	otCliOutputFormat("\r\n\nePSKc : %s", ephemeral_key);
+	otCliOutputFormat("\r\n\nValid for %" PRIu32 " seconds.\r\n", timeout);
+}
+
+void __attribute__((weak)) print_ephemeral_key_expired_message(void)
+{
+	otCliOutputFormat("\r\nEphemeral Key disabled.\r\n");
+}
+
+static otError generate_ephemeral_key(void)
+{
+	otError error = OT_ERROR_NONE;
+	uint8_t i = 0;
+	uint32_t random;
+	char verhoeff_checksum;
+
+	memset(ephemeral_key_string, 0, sizeof(ephemeral_key_string));
+
+	VerifyOrExit(otPlatEntropyGet((uint8_t *)&random, sizeof(random)) == OT_ERROR_NONE,
+		     error = OT_ERROR_FAILED);
+	random %= 100000000;
+	i += snprintf((char *)ephemeral_key_string, sizeof(ephemeral_key_string),
+		      "%08u", random);
+	VerifyOrExit(otVerhoeffChecksumCalculate((const char *)ephemeral_key_string,
+		     &verhoeff_checksum) == OT_ERROR_NONE,
+		     error = OT_ERROR_FAILED);
+	i += snprintf((char *)&ephemeral_key_string[i], sizeof(ephemeral_key_string) - i,
+		      "%c", verhoeff_checksum);
+exit:
+	return error;
+}
+
+static void handle_border_agent_ephemeral_key_callback(void *context)
+{
+	char formatted_epskc[12] = {0};
+	otBorderAgentEphemeralKeyState eph_key_state;
+
+	eph_key_state = otBorderAgentEphemeralKeyGetState((otInstance *)context);
+
+	switch (eph_key_state) {
+	case OT_BORDER_AGENT_STATE_STOPPED:
+		if (epskc_active) {
+			epskc_active = false;
+			print_ephemeral_key_expired_message();
+		}
+		break;
+
+	case OT_BORDER_AGENT_STATE_STARTED:
+		snprintf(formatted_epskc, sizeof(formatted_epskc), "%.3s %.3s %.3s",
+			ephemeral_key_string, ephemeral_key_string + 3,
+			ephemeral_key_string + 6);
```

Comment:
Nit formatting, maybe run `clang-format` on newly added files?
```suggestion
		snprintf(formatted_epskc, sizeof(formatted_epskc), "%.3s %.3s %.3s",
			 ephemeral_key_string, ephemeral_key_string + 3,
			 ephemeral_key_string + 6);
```